Library/Homebrew/download_strategy.rb

```diff
@@ -747,6 +780,40 @@ def fix_absolute_submodule_gitdir_references!
   end
 end
 
+class GitHubGitDownloadStrategy < GitDownloadStrategy
+
+  def initialize(name, resource)
+    super
+    if @url =~ %r{^https?://github\.com/([\w-]+)/([\w.-]+)$}
+      @user = $1
+      @repo = $2
+    end
+  end
+
+  # TODO fix?
+  alias_method :super_last_commit, :last_commit
+
+  # TODO
+  # * rewrite using GitHub module methods
+  # * return if ENV["HOMEBREW_NO_GITHUB_API"] and change the behaviour
+  def last_commit
+    ohai "Fetching last commit from GitHub"
+    args = []
+    args << "https://api.github.com/repos/#{@user}/#{@repo}/commits/#{@ref}"
+    args << "-I" << "-H" << "Accept: application/vnd.github.v3.sha"
+
+    commit = curl_output(*args)[/^ETag: \"(\h+)\"/, 1]
+    commit.slice(0, 7) if commit
```

Comment:
`git rev-parse --short=7 HEAD` does not necessary return string with the length of 7 (It could be more). This makes me wonder should we just store full length commit in the file structure?